### PR TITLE
Allow Nova cloud module to set a specific floating ip address

### PR DESCRIPTION
**Pull request by Enquier**
### What does this PR do?
Add ability of users to specify an floating ip address in the floatingip block in lieu of specifying a pool name

### What issues does this PR fix or reference?
This fixes two things:
1. Allows users to specify a floating ip address in their config files (desirable for machines whose ip might be registered in DNS).
2. Adds a max nova version, recent updates to nova have removed functionality that nova cloud requires. At this time I have tested up to version 6.0.1 for functionality in python-novaclient and added an error for versions greater than that. #41654

### Previous Behavior
1. You could not specify an ip address when assigning a floating ip automatically in the nova cloud module
2. Installing a version of nova newer than 6.0.1 would cause unspecific nova cloud errors

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

**Review by gtmanfred — APPROVED** [review]
I approve this, we need to move away from the nova module.  And locking it in to versions that work for now is ok by me.

File: salt/cloud/clouds/nova.py [hunk 1]
```diff
@@ -709,14 +709,30 @@ def request_instance(vm_=None, call=None):
...
+                        vm_['name'], err
+                    )
+                )
+
+        else:
+            pool = floating_ip_conf.get('pool', 'public')
+            for fl_ip, opts in six.iteritems(conn.floating_ip_list()):
+                if opts['fixed_ip'] is None and opts['pool'] == pool:
+                    floating_ip = fl_ip
+                    log.debug
```
Comment (cachedout) [inline salt/cloud/clouds/nova.py]: This line should either be fixed or removed.

**Review by cachedout — CHANGES_REQUESTED** [review]

File: salt/cloud/clouds/nova.py (hunk 1 above)
Comment (Enquier) [inline salt/cloud/clouds/nova.py]: Line removed in latest commit

**Review by cachedout — APPROVED** [review]

**Comment by cachedout** [comment]
Merged. Thanks very much, @Enquier !

**Pull request merged**